app-header-layout/app-header-layout.html

```diff
@@ -168,10 +186,12 @@
         var headerHeight = header.offsetHeight;
         // Update the header position.
         if (!this.hasScrollingRegion) {
-          var rect = this.getBoundingClientRect();
-          var rightOffset = document.documentElement.clientWidth - rect.right;
-          header.style.left = rect.left + 'px';
-          header.style.right = rightOffset + 'px';
+          requestAnimationFrame(function() {
```

Comment:
I've been using demo/demo2.html with Polymer 2. Note that this only happens when there is exactly 1 resize event between wide and narrow (e.g. rotating a phone, or toggling the dev tools pane).